File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s196249385_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	parts := strings.Split(scanner.Text(), " ")
	n, _ := strconv.Atoi(parts[0])
	k, _ := strconv.Atoi(parts[1])

	scanner.Scan()
	aStrs := strings.Split(scanner.Text(), " ")
	a := make([]int64, n)
	for i := 0; i < n; i++ {
		a[i], _ = strconv.ParseInt(aStrs[i], 10, 64)
	}

	cumsum := make([]int64, n+1)
	for i := 0; i < n; i++ {
		cumsum[i+1] = cumsum[i] + a[i]
	}

	b := make([]int64, 0)
	for i := 0; i < n; i++ {
		for j := i; j < n; j++ {
			sum := cumsum[j+1] - cumsum[i]
			b = append(b, sum)
		}
	}

	var max int64 = 0
	for _, e := range b {
		if e > max {
			max = e
		}
	}

	mask := int64(1)
	for mask <= max {
		mask <<= 1
	}
	mask >>= 1

	for mask > 0 {
		var tmp []int64
		for _, e := range b {
			if (e & mask) != 0 {
				tmp = append(tmp, e)
			}
		}
		if len(tmp) >= k {
			b = tmp
		}
		mask >>= 1
	}

	if len(b) < k {
		fmt.Println(0)
	} else {
		ans := int64(-1)
		for _, e := range b {
			ans &= e
		}
		fmt.Println(ans)
	}
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Python/s196249385_commented.py
n, k = map(int, input().split())
a = list(map(int, input().split()))
cumsum = [0] * (n + 1)
for i in range(n):
    cumsum[i + 1] = cumsum[i] + a[i]

b = []
max_sum = 0
for i in range(n):
    for j in range(i, n):
        s = cumsum[j + 1] - cumsum[i]
        b.append(s)
        if s > max_sum:
            max_sum = s

mask = 0
if max_sum > 0:
    mask = 1 << (max_sum.bit_length() - 1)

while mask > 0:
    tmp = []
    for e in b:
        if (e & mask) != 0:
            tmp.append(e)
    if len(tmp) >= k:
        b = tmp
    mask >>= 1

if len(b) < k:
    print(0)
else:
    ans = -1
    for e in b:
        ans &= e
    print(ans)
# End of Code
